src/cdpTargetsProvider.ts

```diff
@@ -70,4 +87,57 @@ export default class CDPTargetsProvider implements vscode.TreeDataProvider<CDPTa
         this.telemetryReporter.sendTelemetryEvent("view/refresh");
         this.changeDataEvent.fire(null);
     }
+
+    private downloadFaviconFromSitePromise(url: string) : Promise<string | null> | null {
+        if (!this.extensionPath || !url) {
+            return null;
+        }
+        const https = require('https');
+        const fs = require('fs');
+        const faviconRegex = /((?:\/\/|\.)([^\.]*)\.[^\.^\/]+\/).*/;
+
+        // Example regex match: https://docs.microsoft.com/en-us/microsoft-edge/
+        // urlMatch[0] = .microsoft.com/en-us/microsoft-edge/
+        // urlMatch[1] = .microsoft.com/
+        // urlMatch[2] = microsoft
+        const urlMatch = url.match(faviconRegex);
+        let filename;
+        if (urlMatch) {
+            filename = `${urlMatch[2]}Favicon.ico`;
+        } else {
+            return null;
+        }
+
+        // Replacing ".microsoft.com/en-us/microsoft-edge/" with ".microsoft.com/favicon.ico"
+        const faviconUrl = url.replace(faviconRegex, "$1favicon.ico");
+
+        const filePath = path.join(this.extensionPath, "resources", "favicons", filename);
+
+        const file = fs.createWriteStream(filePath);
+        const promise = new Promise<string | null>((resolve, reject) => {
+            https.get(faviconUrl, (response: any) => {
+                response.pipe(file);
+                file.on('error', () => {
+                    resolve(null);
+                });
+                file.on('finish', () => {
+                    if (file.bytesWritten) {
+                        resolve(filePath);
+                    } else {

```

Comment:
how and when are we removing this?

Reply:
Added `clearFaviconResourceDirectory()` call to refresh.  It will clear the `resources/favicon` directory each time, so only the currently used icons will be present in the folder.